trio/_core/_ki.py

```diff
@@ -177,7 +178,7 @@ def wrapper(*args, **kwargs):
 @contextmanager
 def ki_manager(deliver_cb, restrict_keyboard_interrupt_to_checkpoints):
     if (
-        threading.current_thread() != threading.main_thread()
+        not is_main_thread()
```

Comment:
Can you grep for `current_thread`? I think there are 3-4 other places where we use this check and should switch to this. (Definitely in `catch_signals` and `_io_windows.py` at least.)